packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -141,36 +145,105 @@ export class DefaultHasManyThroughRepository<
       throughData?: DataObject<ThroughEntity>;
       throughOptions?: Options;
     },
-  ): Promise<void> {
-    throw new Error('Method not implemented.');
+  ): Promise<TargetEntity> {
+    const targetRepository = await this.getTargetRepository();
+    const throughRepository = await this.getThroughRepository();
+    const targetInstance = await targetRepository.create(
+      targetModelData,
+      options,
+    );
+    const targetConstraint = this.getThroughFkConstraint(targetInstance);
+    const throughConstraint = this.getThroughConstraint();
+    const constraints = {...targetConstraint, ...throughConstraint};
+    await throughRepository.create(
+      constrainDataObject(
+        options?.throughData ?? {},
+        constraints as DataObject<ThroughEntity>,
+      ),
+      options?.throughOptions,
+    );
+    return targetInstance;
   }
 
   async find(
     filter?: Filter<TargetEntity>,
     options?: Options & {
       throughOptions?: Options;
     },
-  ): Promise<void> {
-    throw new Error('Method not implemented.');
+  ): Promise<TargetEntity[]> {
+    const targetRepository = await this.getTargetRepository();
+    const throughRepository = await this.getThroughRepository();
+    const throughConstraint = this.getThroughConstraint();
+    const throughInstances = await throughRepository.find(
+      constrainFilter(undefined, throughConstraint),
+      options?.throughOptions,
+    );
+    const targetConstraint = this.getTargetConstraint(throughInstances);
+    return targetRepository.find(
+      constrainFilter(filter, targetConstraint),
+      options,
+    );
   }
 
+  // 1. Given through links: {sourceId: 1, targetId: 1} and {sourceId: 2, targetId: 2}.
+  // When delete all models belonging to source id 1,
+  // then the target model id 1 and the first link is deleted.
+
+  // 2. Given trough links {sourceId: 1, targetId: 1}, {sourceId: 2, targetId: 1}
+  // and {sourceId: 2, targetId: 2}.
+  // When delete all models belonging to source id 1,
+  // then the target model id 1 and the first two links are deleted.
+  // Only {sourceId: 2, targetId: 2} link is preserved
   async delete(
     where?: Where<TargetEntity>,
     options?: Options & {
       throughOptions?: Options;
     },
   ): Promise<Count> {
-    throw new Error('Method not implemented.');
-  }
+    const targetRepository = await this.getTargetRepository();
+    const throughRepository = await this.getThroughRepository();
+    const throughConstraint = this.getThroughConstraint();
+    const throughInstances = await throughRepository.find(
+      constrainFilter(undefined, throughConstraint),
```

Comment:
hmm, why the first param is `undefined`?

Reply:
Because there is no original filter to apply to the through repo. Here's the API docs:
```
export function constrainFilter<T extends object>(
  originalFilter: Filter<T> | undefined,
  constraint: AnyObject,
): Filter<T> 
...
```
I think either `{}` or `undefined` works.